core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -2278,6 +2377,62 @@ public void _run() throws IOException, InstallationStatus {
             }
         }
 
+        /**
+         * If we happen to have the file already in the {@coode WEB-INF/detached-plugins} directory and it happens to
+         * match the checksum we were expecting, then save ourselves a trip to the download site. This method is
+         * best-effort, and if anything goes wrong we simply fall back to the standard download path.
+         *
+         * @return The cached file, or null for a cache miss
+         */
+        @CheckForNull
+        private File getCached(DownloadJob job) {
+            URL src;
+            try {
+                /*
+                 * Could make PluginManager#getDetachedLocation public and consume it here, but this method is
+                 * best-effort anyway.
+                 */
+                src = Jenkins.get().servletContext.getResource(String.format("/WEB-INF/detached-plugins/%s.hpi", plugin.name));
```

Comment:
FWIW the `detached-plugins` folder is customizable and CloudBees customizes this even outside of FIPS.

https://github.com/jenkinsci/jenkins/blob/b2021f2464b361d4afa00cf289dbcd1fd7c5f051/core/src/main/java/hudson/PluginManager.java#L640-L647

Reply:
> FWIW

Is worth of this feedback affected by the fact that I already discussed this in the comment a few lines above?

Reply:
> and CloudBees customizes this even outside of FIPS

Can you point me to the part of your comment that says that please because I must have missed it when I read it.